packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -507,8 +509,42 @@ export interface CommonProjectProps {
    * The physical, human-readable name of the CodeBuild Project.
    */
   projectName?: string;
-}
 
+  /**
+   * VPC network to place codebuild network interfaces
+   *
+   * Specify this if the codebuild project needs to access resources in a VPC.
+   */
+  vpc?: ec2.IVpcNetwork;
+
+  /**
+   * Where to place the network interfaces within the VPC.
+   *
+   * Only used if 'vpc' is supplied.
+   *
+   * @default All private subnets
+   */
+  vpcPlacement?: ec2.VpcPlacementStrategy;
+
+  /**
+   * What security group to associate with the codebuild project's network interfaces.
+   *
+   * Only used if 'vpc' is supplied.
+   *
+   */
+  securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * Whether to allow the CodeBuild to send all network traffic
+   *
+   * If set to false, you must individually add traffic rules to allow the
+   * CodeBuild project to connect to network targets.
+   *
+   * @default true
+   */
+  allowAllOutbound?: boolean;
```

Comment:
I would underscore that this is only in effect when `vpc` has been provided, like you do for the other new properties in this interface.